vpd-tool/src/vpd_tool.cpp

```diff
@@ -8,13 +14,60 @@ int VpdTool::readKeyword(const std::string& i_fruPath,
                          const bool i_onHardware,
                          const std::string& i_fileToSave)
 {
-    // ToDo: Need to add implementation details
-    (void)i_fruPath;
-    (void)i_recordName;
-    (void)i_keywordName;
-    (void)i_onHardware;
-    (void)i_fileToSave;
-
-    return 0;
+    int l_rc = -1;
+    try
+    {
+        types::DbusVariantType l_keywordValue;
+        if (i_onHardware)
+        {
+            // TODO: Implement read keyword's value from hardware
+        }
+        else
+        {
+            std::string l_inventoryObjectPath(constants::baseInventoryPath +
+                                              i_fruPath);
+
+            l_keywordValue = utils::readDbusProperty(
+                constants::inventoryManagerService, l_inventoryObjectPath,
+                constants::ipzVpdInf + i_recordName, i_keywordName);
+        }
+
+        if (const auto l_value =
+                std::get_if<types::BinaryVector>(&l_keywordValue))
+        {
+            const std::string& l_keywordStrValue =
+                utils::getPrintableValue(*l_value);
```

Comment:
As part of code refactoring, we have decided to print value in both the formats - HEX and ASCII. Please revisit on this

Reply:
if the value is printable then value is printed in ASCII format,
if not value is printed in hex format.